pandas/core/series.py

```diff
@@ -166,9 +166,9 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                     else:
                         index = Index(_try_sort(data))
                 try:
-                    if isinstance(index, DatetimeIndex):
+                    if isinstance(index, DatetimeIndex) and lib.infer_dtype(data) != 'datetime64':
```

Comment:
The idea is to have the least special cases as possible. You rarely actually want to convert a ``DatetimeIndex`` using ``.astype('O')`` this is the least desirable result (though it may have been done for some reason). But it is tricky because a user can pass ``Timestamp/datetime.datetime/datetime.date/np.datetime64`` as elements and a non-index like (but that is index-like). Best best is to do ``_ensure_index`` which will convert list-like things to an ``Index`` (and possibly to a ``DatetimeIndex`` if its compat; if its not then all bets are off).